### PR TITLE
Fix tests that are failing after introduction of ipam factories

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes: #2689
# What's Changed

Fix some more tests that were failing that weren't identified in the original factory fixture PRs

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- n/a Attached Screenshots, Payload Example
- n/a Unit, Integration Tests
- n/a Documentation Updates (when adding/changing features)
- n/a Example Plugin Updates (when adding/changing features)
- n/a Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/tests/test_filters.py [hunk 1]
```diff
@@ -1401,10 +1402,17 @@ def test_color(self):
 
     def test_search(self):
         params = {"q": "active"}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 1)
-        value = self.queryset.values_list("pk", flat=True)[0]
+        q = Q(name__icontains="active") | Q(slug__icontains="active") | Q(content_types__model__icontains="active")
```
Comment (glennmatthews) [inline nautobot/extras/tests/test_filters.py:1405]: I assume the content-types filter is just matching the `search()` implementation? We don't actually have any models at present that match this, right?

File: nautobot/virtualization/tests/test_filters.py [hunk 2]
```diff
@@ -638,22 +640,40 @@ def test_mode(self):
 
     def test_ip_addresses(self):
```
Comment (glennmatthews) [inline nautobot/virtualization/tests/test_filters.py:641]: Are these changes fixing #2689?

File: nautobot/virtualization/tests/test_filters.py [hunk 3]
```diff
@@ -638,22 +640,40 @@ def test_mode(self):
...
         with self.subTest("Primary Addresses"):
-            ipaddress = IPAddress.objects.last()
-            params = {"ip_addresses": ["192.0.2.1/24", ipaddress.id]}
-            self.assertEqual(self.filterset(params, self.queryset).qs.count(), 2)
+            vminterface_ct = ContentType.objects.get_for_model(VMInterface)
+            ipaddresses = list(
+                IPAddress.objects.filter(assigned_object_id__isnull=False, assigned_object_type=vminterface_ct.pk)[:2]
```
Comment (glennmatthews) [inline nautobot/virtualization/tests/test_filters.py:645]: I think?

```suggestion
                IPAddress.objects.filter(assigned_object_id__isnull=False, assigned_object_type=vminterface_ct)[:2]
```

**Review by glennmatthews — APPROVED** [review]

File: nautobot/virtualization/tests/test_filters.py (hunk 2 above)
Comment (HanlinMiao) [reply nautobot/virtualization/tests/test_filters.py:641]: Here is the change I made that seems to fix the error on my end. https://github.com/nautobot/nautobot/commit/6ab61c3e795480ba76f417d7a1b7808d1784aa37

File: nautobot/extras/tests/test_filters.py (hunk 1 above)
Comment (gsnider2195) [reply nautobot/extras/tests/test_filters.py:1405]: It shouldn't match on anything but we can't say for sure anymore now that the test fixtures are being randomly generated by factories. This should be the same Q object that's generated by the search filter's `generate_query` method but I didn't want to leverage a method in the filterset to test the filterset.

File: nautobot/virtualization/tests/test_filters.py (hunk 3 above)
Comment (gsnider2195) [reply nautobot/virtualization/tests/test_filters.py:645]: That works!

File: nautobot/virtualization/tests/test_filters.py (hunk 2 above)
Comment (gsnider2195) [reply nautobot/virtualization/tests/test_filters.py:646]: > Are these changes fixing #2689?

They are. Thanks for the reminder

**Pull request merged**